From d405e39901429aadd49e67096c5ba04945ab396d Mon Sep 17 00:00:00 2001
From: Yannick Lanz <yannick.lanz@wifx.net>
Date: Wed, 22 Sep 2021 11:45:42 +0200
Subject: [PATCH 8/9] l1: re-enable tcb timer (only one) and disable tclib to
 keep highres timer

Revert previous commit without boot log error message.

Signed-off-by: Yannick Lanz <yannick.lanz@wifx.net>
---
 arch/arm/boot/dts/l1.dts      | 13 +++++++++++--
 arch/arm/configs/l1_defconfig |  2 --
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/l1.dts b/arch/arm/boot/dts/l1.dts
index 462077219d74..006b2fd851f7 100644
--- a/arch/arm/boot/dts/l1.dts
+++ b/arch/arm/boot/dts/l1.dts
@@ -250,10 +250,19 @@
 				status = "okay";
 			};
 
-			/* disable unused TCB */
 			tcb0: timer@f801c000 {
-				status = "disabled";
+				timer0: timer@0 {
+					compatible = "atmel,tcb-timer";
+					reg = <0>;
+				};
+
+				timer1: timer@1 {
+					compatible = "atmel,tcb-timer";
+					reg = <1>;
+				};
 			};
+
+			/* disable unused TCB */
 			tcb1: timer@fc020000 {
 				status = "disabled";
 			};
diff --git a/arch/arm/configs/l1_defconfig b/arch/arm/configs/l1_defconfig
index 1eebced7d5f9..57cacab65b1f 100644
--- a/arch/arm/configs/l1_defconfig
+++ b/arch/arm/configs/l1_defconfig
@@ -20,7 +20,6 @@ CONFIG_BLK_DEV_INITRD=y
 CONFIG_EMBEDDED=y
 CONFIG_ARCH_AT91=y
 CONFIG_SOC_SAMA5D4=y
-# CONFIG_ATMEL_CLOCKSOURCE_TCB is not set
 CONFIG_FORCE_MAX_ZONEORDER=15
 CONFIG_UACCESS_WITH_MEMCPY=y
 # CONFIG_ATAGS is not set
@@ -70,7 +69,6 @@ CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=4
 CONFIG_BLK_DEV_RAM_SIZE=8192
-CONFIG_ATMEL_TCLIB=y
 CONFIG_ATMEL_SSC=y
 CONFIG_EEPROM_AT24=y
 CONFIG_EEPROM_93CX6=m
-- 
2.25.1

